app/controllers/AFTOverviewController.scala

```diff
@@ -0,0 +1,141 @@
+
+class AFTOverviewController @Inject()(
+                                       identify: IdentifierAction,
+                                       renderer: Renderer,
+                                       allowAccess: AllowAccessActionProviderForIdentifierRequest,
+                                       config: FrontendAppConfig,
+                                       schemeService: SchemeService,
+                                       quartersService: QuartersService,
+                                       paymentsAndChargesService: PaymentsAndChargesService,
+                                       override val messagesApi: MessagesApi,
+                                       val controllerComponents: MessagesControllerComponents
+                                     )(implicit ec: ExecutionContext)
+  extends FrontendBaseController with I18nSupport with NunjucksSupport {
+
+  def onPageLoad(srn: String): Action[AnyContent] = (identify andThen allowAccess(Some(srn))).async {
+    implicit request =>
+      (for {
+        outstandingAmount <- getOutstandingPaymentAmount(srn)
+        schemeDetails <- schemeService.retrieveSchemeDetails(psaId = request.idOrException, schemeIdType = schemeIdType, srn = srn)
+        quartersInProgress <- quartersService.getInProgressQuarters(srn = srn, pstr = schemeDetails.pstr)
+        allPastYears <- quartersService.getPastYears(pstr = schemeDetails.pstr)
+        pastYearsAndQuarters <- Future.traverse(displayYears(allPastYears))(
+          year => quartersService.getPastQuarters(pstr = schemeDetails.pstr, year = year).flatMap(quarters => Future.successful(year, quarters))
+        )
+      }
+      yield OverviewInfo(schemeDetails, quartersInProgress, pastYearsAndQuarters, outstandingAmount)
+        ).flatMap { overviewInfo =>
+        renderer.render(aftOverviewTemplate,
+          Json.obj(viewModel ->
+            OverviewViewModel(
+              returnUrl = config.schemeDashboardUrl(request).format(srn),
+              newAftUrl = controllers.routes.YearsController.onPageLoad(srn).url,
+              paymentsAndChargesUrl = controllers.financialOverview.scheme.routes.SelectYearController.onPageLoad(srn, AccountingForTaxCharges).url,
+              schemeName = overviewInfo.schemeDetails.schemeName,
+              outstandingAmount = overviewInfo.outstandingAmount,
+              quartersInProgress = overviewInfo.quartersInProgress.map(q => textAndLinkForQuarter(formatForDisplayOneYear, q, srn)),
+              pastYearsAndQuarters = overviewInfo.pastYearsAndQuarters.map(pYAQ => (pYAQ._1, pYAQ._2.map(q => textAndLinkForQuarter(monthDayStringFormat, q, srn)))),
+              viewAllPastAftsUrl = controllers.routes.PastAftReturnsController.onPageLoad(srn, 0).url
+            )
+          )
+        ).map(Ok(_))
+      }.recover {
+        case e: Exception =>
+          logger.error("Failed to retrieve scheme details or past year details", e)
+          InternalServerError("An error occurred")
```

Comment:
This error will return text only page. This should either redirect to an error page or return an error view.